runtime-components/non-repudiation-postgresql/src/test/scala/com/daml/nonrepudiation/postgresql/NonRepudiationProxyConformance.scala

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.postgresql
+
+import java.security.KeyPair
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.{ExecutorService, Executors, ThreadFactory}
+
+import cats.effect.{Blocker, ContextShift, IO}
+import com.daml.ledger.api.testtool.infrastructure.{
+  LedgerTestCasesRunner,
+  LedgerTestSummary,
+  Result,
+}
+import com.daml.ledger.api.testtool.suites.ClosedWorldIT
+import com.daml.ledger.api.testtool.tests._
+import com.daml.ledger.api.v1.command_service.CommandServiceGrpc.CommandService
+import com.daml.ledger.api.v1.command_submission_service.CommandSubmissionServiceGrpc.CommandSubmissionService
+import com.daml.ledger.resources.{ResourceContext, ResourceOwner}
+import com.daml.nonrepudiation.client.SigningInterceptor
+import com.daml.nonrepudiation.{AlgorithmString, NonRepudiationProxy}
+import com.daml.platform.sandbox.config.SandboxConfig
+import com.daml.platform.sandboxnext.{Runner => Sandbox}
+import com.daml.ports.Port
+import com.daml.testing.postgresql.PostgresAroundAll
+import com.zaxxer.hikari.HikariDataSource
+import doobie.hikari.HikariTransactor
+import doobie.util.log.LogHandler
+import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc.netty.NettyChannelBuilder
+import org.scalatest.EitherValues
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.slf4j.LoggerFactory
+import sun.security.tools.keytool.CertAndKeyGen
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.DurationInt
+
+final class NonRepudiationProxyConformance
+    extends AsyncFlatSpec
+    with Matchers
+    with EitherValues
+    with PostgresAroundAll {
```

Comment:
Should we have an in-memory backend (e.g. h2) for testing (by users) and debugging?

Reply:
Should we simply expose the map-backed implementation of the repositories or did you have something else in mind?

Reply:
map-backed seems perfectly fine. Not sure if it’s easier to keep that working or to have an h2 implementation. happy to leave it up to you.

Reply:
I would keep the map-backed implementation where it is right now, it should be fine for testing without adding development overhead. We can easily revise this decision in the future. But thanks for raising this!